DuckDuckGo/TabViewController.swift

```diff
@@ -1529,6 +1560,23 @@ extension TabViewController: WKNavigationDelegate {
         }
     }
 
+    private func inferLoadContex(for navigationAction: WKNavigationAction) {
+        if navigationAction.navigationType != .reload {
+            if let currentUrl = webView.url, currentUrl.isDuckDuckGoSearch,
+               let newUrl = navigationAction.request.url, !newUrl.isDuckDuckGoSearch {
```

Comment:
lets refactor these two ifs into guards as follows:

guard navigationAction.navigationType != .reload else {
            return nil
        }
        
        guard let currentUrl = webView.url, 
              let newUrl = navigationAction.request.url else {
            return nil
        }

and then:

if currentUrl.isDuckDuckGoSearch && !newUrl.isDuckDuckGoSearch {
            return .serp
        }

        switch navigationAction.navigationType {
        case .linkActivated, .other, .formSubmitted:
            return .navigation
        default:
            return nil
        }